### PR TITLE
🐛 Fix kustomize_substitutions

### LINKED ISSUE #3375: kustomize_substitutions in tilt-settings.json is broken
**What steps did you take and what happened:**
1. Try to use `kustomize_substitutions` for variables that you do **not** set as environment variables
1. These placeholders are replaced with the empty string

**What did you expect to happen:**
1. Successful replacement from `kustomize_substitutions` from `tilt-settings.json`

**Anything else you would like to add:**
The change to use kustomize with envsubst (#3361) accidentally broke this. We can maybe try to copy all the substitutions into `os.environ` before running kustomize.

/kind bug

### PR DESCRIPTION
**Pull request by ncdc**
**What this PR does / why we need it**:
Copy kustomize_substitutions into os.environ before calling kustomize +
envsubst to ensure the values are available to envsubst.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3375

### CODE REVIEW

File: Tiltfile [hunk 1]
```diff
@@ -190,17 +190,16 @@ def enable_provider(name):
...
-    # Apply the kustomized yaml for this provider
-    yaml = str(kustomize_with_envsubst(context + "/config"))
+    # Copy all the substitutions from the user's tilt-settings.json into the environment. Otherwise, the substitutions
+    # are not available and their placeholders will be replaced with the empty string when we call kustomize +
+    # envsubst below.
     substitutions = settings.get("kustomize_substitutions", {})
     for substitution in substitutions:
-        value = substitutions[substitution]
-        yaml = yaml.replace("${" + substitution + "}", value)
+        os.environ[substitution] = substitutions[substitution]
```
Comment (alexeldeib) [inline Tiltfile:198]: can use os.environ.update(substitutions) to simplify a bit

**Comment by alexeldeib** [comment]
didn't realize you self assigned, was about to open the same: https://github.com/kubernetes-sigs/cluster-api/compare/master...alexeldeib:ace/tiltish?expand=1#diff-1a474bf970407efc3332970120c1a429R195

File: Tiltfile (hunk 1 above)
Comment (ncdc) [reply Tiltfile:198]: Thanks, I was looking at Tilt's docs on the Starlark syntax and didn't see that. Will make the change.

**Comment by ncdc** [comment]
Updated, PTAL!

**Pull request merged**